orchestrator/relayer/src/fee_manager.rs

```diff
@@ -0,0 +1,211 @@
+use gravity_utils::types::{ Erc20Token, };
+use std::collections::HashMap;
+use std::time::Duration;
+use std::time::Instant;
+use gravity_utils::types::config::RelayerMode;
+use serde_json::Map;
+use reqwest::Error;
+
+pub struct FeeManager {
+    token_price_map: Map<String, serde_json::Value>,
+    token_api_path_map: Map<String, serde_json::Value>,
+    next_batch_send_time: HashMap<EthAddress, Instant>,
+    mode: RelayerMode,
+}
+
+#[derive(serde::Deserialize, Debug)]
+struct ApiResponse {
+    status: String,
+    result: ApiResult,
+}
+
+#[derive(serde::Deserialize, Debug)]
+struct ApiResult {
+    token_pair_decimal: u32,
+    aggregated_price: f64,
+}
+
+impl FeeManager {
+    pub async fn new_fee_manager(mode: RelayerMode) -> FeeManager {
+        let mut fm =  Self {
+            token_price_map: Default::default(),
+            token_api_path_map: Default::default(),
+            next_batch_send_time: HashMap::new(),
+            mode
+        };
+
+        fm.init()
+            .await;
+        return fm;
+    }
+
+    async fn init(&mut self) {
+        match self.mode {
+            RelayerMode::Api => {
+                let token_addresses_path =
+                    std::env::var("TOKEN_ADDRESSES_JSON").unwrap_or_else(|_| "token_addresses.json".to_owned());
+
+                let token_addresses_str = match tokio::fs::read_to_string(token_addresses_path).await {
+                    Err(err) => {
+                        panic!("Error while fetching token pair addresses {}", err);
+                    }
+                    Ok(value) => value,
+                };
+
+                let token_addresses: serde_json::Map<String, serde_json::Value> = match serde_json::from_str(&token_addresses_str)
+                {
+                    Err(err) => {
+                        panic!("Error while parsing token pair addresses json configuration: {}", err);
+                    }
+                    Ok(token_addresses) => token_addresses,
+                };
+
+                self.token_api_path_map = token_addresses;
+            }
+            RelayerMode::File => {
+                let config_file_path =
+                    std::env::var("TOKEN_PRICES_JSON").unwrap_or_else(|_| "token_prices.json".to_owned());
+
+                let config_str = match tokio::fs::read_to_string(config_file_path).await {
+                    Err(err) => {
+                        panic!("Error while fetching token prices {}", err);
+                    }
+                    Ok(value) => value,
+                };
+
+                let config: serde_json::Map<String, serde_json::Value> = match serde_json::from_str(&config_str)
+                {
+                    Err(err) => {
+                        panic!("Error while parsing token prices json configuration: {}", err);
+                    }
+                    Ok(config) => config,
+                };
+
+                self.token_price_map = config;
+            }
+            _ => {}
```

Comment:
Just curious, do we need to consider the 'AlwaysRelay' case for this match?

Reply:
yes good catch. should return true there